### ISSUE TITLE
Sparese techniques

### ISSUE THREAD

**Issue by Jingfei-Liu** [jonathf/chaospy#191]
Hi Jonathan：

Thanks for releasing such an amazing software. I am wondering if there are some sparse techniques available in chaospy, like least angle regression, compressive sensing and so on.

Best regards!

**Comment by jonathf** [comment]
It did once upon a time, but decided to remove it in favor of having an exposed attributes to do so using other libraries, so to not reinvent the wheel.

In this case, you want to do the regression part of polynomial chaos expansion creation using something like `scikit-learn`. For exanple for least angle regression, this can be done as follows:
```python
samples = dist.sample(N)
model_evals = [foo(sample) for sample in samples.T]
poly_evals = polynomials(*samples).T

model = sklearn.linear_model.Lars(fit_intercept=False, normalize=False)
model.fit(poly_evals, model_evels)
uhat = model.coeffs_
approx_model = chaospy.sum(polynomials*uhat, -1)
```

Be aware that scikit-learn sometimes does extra stuff like `fit_intercept`, `normalize` etc. that disturbs the direct relationship between the `.fit` input and the `.coeffs_` output.

Also, please experiment for yourself. I haven't tested the code above, nor done interactions between chaospy and sklearn in while, to be certain the code above works as is.

**Comment by jonathf** [comment]
I have made Lars work before, but my experience with compressed sensing is limited. If you get a working example working, I'd love to take a look.

**Comment by Jingfei-Liu** [comment]
Thank you for such patient answer, it helps me a lot, I will try it myself. 
Yet, I am a beginner for PCE, I am looking forward to find some good resource to learn.
The matlab code for compressed sensing assisted PCE can be found in the UQlab at https://www.uqlab.com/ , and it also appears in PyApprox at https://github.com/sandialabs/pyapprox.
Hope it can make a little help.

**Comment by jonathf** [comment]
Okay. Thanks for the links.
I am aware of UQlab, but PyApprox is new to me.

I am curious. How did you get to know PyApprox? It is only 18 days old on Github. Did you just stumble upon it?

**Comment by Jingfei-Liu** [comment]
Yes, I found PyApprox when I was searching some  key words around polynomial chaos expension on the Github.
